dali/pipeline/operators/util/pad_test.cc

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/pipeline/data/tensor.h"
+#include "dali/test/dali_operator_test.h"
+#include "dali/test/dali_operator_test_utils.h"
+
+namespace dali {
+
+namespace testing {
+
+// 1-d elements: {{5},{2},{3}} -> {{5}, {5}, {5}}
+std::vector<std::vector<float> > batch1 = {{3, 4, 2, 5, 4},
+                                           {2, 2},
+                                           {3, 199, 5}};
+std::vector<std::vector<float> > padded_batch1 = {{3, 4, 2, 5, 4},
+                                                  {2, 2, -1, -1, -1},
+                                                  {3, 199, 5, -1, -1}};
+
+// 2-d elements: {{2, 4}, {2, 2}} -> {{2, 4}, {2, 4}}
+std::vector<std::vector<int> > batch2 = {{1, 2 , 3, 4,
+                                          5, 6, 7, 8},
+                                         {1, 2,
+                                          4, 5}};
+std::vector<std::vector<int> > padded_batch2 = {{1, 2 , 3, 4,
+                                                 5, 6, 7, 8},
+                                                {1, 2, 42, 42,
+                                                 4, 5, 42, 42}};
+
+template <typename T>
+std::vector<std::vector<T> > GetPaddedBatchForAxis(int axis) {
+  return {};
+}
+
+template <>
+std::vector<std::vector<float> > GetPaddedBatchForAxis(int axis) {
```

Comment:
I think those functions are confusing the reader more than anything else. If we only have those hardcoded values I guess it would be simpler to just use those global variables directly in the test

Reply:
The thing is that PadVerify is templated and `GetPaddedBatchForAxis` return type depends of the template. How would you do without these functions ? (+ I tried to add some flexibility to add more test cases)

Reply:
OK I get it. I think it can stay like this